pkg/cmd/repo/fork/fork.go

```diff
@@ -0,0 +1,292 @@
+package fork
+
+import (
+	"errors"
+	"fmt"
+	"net/http"
+	"net/url"
+	"strings"
+	"time"
+
+	"github.com/cli/cli/api"
+	"github.com/cli/cli/context"
+	"github.com/cli/cli/git"
+	"github.com/cli/cli/internal/config"
+	"github.com/cli/cli/internal/ghrepo"
+	"github.com/cli/cli/internal/run"
+	"github.com/cli/cli/pkg/cmdutil"
+	"github.com/cli/cli/pkg/iostreams"
+	"github.com/cli/cli/pkg/prompt"
+	"github.com/cli/cli/utils"
+	"github.com/spf13/cobra"
+)
+
+type ForkOptions struct {
+	HttpClient func() (*http.Client, error)
+	Config     func() (config.Config, error)
+	IO         *iostreams.IOStreams
+	BaseRepo   func() (ghrepo.Interface, error)
+	Remotes    func() (context.Remotes, error)
+	Since      func(t time.Time) time.Duration
+
+	Repository   string
+	Clone        bool
+	Remote       bool
+	PromptClone  bool
+	PromptRemote bool
+}
+
+// TODO decide if this should be injected via a factory.
+var Since = func(t time.Time) time.Duration {
+	return time.Since(t)
```

Comment:
I don't think there is need to allow passing Since via factory just yet, as we can easily stub it via ForkOptions for tests where it matters.

However, when it comes to `ForkOptions.Since`, I wonder whether it would make sense to name it `ForkOptions.Now` with the value of `time.now()` at runtime and a fixed time in tests? We solve the same problem of freezing time, but in a more general way

Reply:
I specifically wanted to use the standard library's `Since` function and I would prefer to stub that out instead of stubbing `Now` and doing the math ourselves.

The field in `ForkOptions` is actually vestigial -- since we're just porting the existing test there is no split between testing argument processing and testing behavior.

For now I'm going to take out the `ForkOptions` field and when we eventually re-do the tests for fork we can revisit.